src/views/virtualmachines/utils/virtualMachineStatuses.ts

```diff
@@ -36,33 +37,29 @@ export const errorPrintableVMStatus = {
 export const isFailedPrintableStatus = (printableStatus: string) =>
   Object.values(errorPrintableVMStatus).includes(printableStatus);
 
-export const getVMStatusIcon = (
-  status: string,
-  //   arePendingChanges: boolean,
-): React.ComponentClass | React.FC => {
-  let icon: React.ComponentClass | React.FC = UnknownIcon;
-
+export const getVMStatusIcon = (status: string): React.ComponentClass | React.FC => {
   if (status === printableVMStatus.Paused) {
-    icon = PausedIcon;
-  } else if (status === printableVMStatus.Running) {
-    icon = SyncAltIcon;
-  } else if (status === printableVMStatus.Stopped) {
-    icon = OffIcon;
-  } else if (status?.toLowerCase().includes('error')) {
-    icon = ExclamationCircleIcon;
-  } else if (
-    status?.toLowerCase().includes('pending') ||
-    status?.toLowerCase().includes('provisioning')
-  ) {
+    return PausedIcon;
+  }
+  if (status === printableVMStatus.Running) {
+    return SyncAltIcon;
+  }
+  if (status === printableVMStatus.Stopped) {
+    return OffIcon;
+  }
+  if (status === printableVMStatus.Migrating) {
+    return MigrationIcon;
+  }
+  if (status?.toLowerCase().includes('error')) {
```

Comment:
This won't catch `ErrImagePull`. Why not make this a switch statement?

Reply:
you're right, this logic was copied from 4.10 where we did things a bit different, changed to switch case and to catch all error statuses, thanks :)